native/cocos/core/scene-graph/Node.cpp

```diff
@@ -886,6 +886,16 @@ void Node::clearNodeArray() {
     }
 }
 
+ccstd::string Node::getPathInHierarchy() const {
+    ccstd::string result = getName();
+    Node *curNode = getParent();
+    while (curNode && dynamic_cast<Scene *>(curNode) == nullptr) {
```

Comment:
dynamic_cast<Scene *>(curNode) == nullptr -> dynamic_cast<Scene *>(curNode) != nullptr

Reply:
As I can see, the original idea in TS is to judge curNode not a Scene @Zhmz

Reply:
If so, then its parent is null, don't have to check type. Please double check the meaning of it.

Reply:
using "parent == nullptr" can exactly determine that the curNode is a scene type.
All comman nodes have their own parent but scenes don't have.

Reply:
> If so, then its parent is null, don't have to check type. Please double check the meaning of it.

Yes. You're right. Wait for @Zhmz to make sure.